src/algebraic_geometry/structure_sheaf.lean

```diff
@@ -328,33 +340,265 @@ def stalk_iso_Type (x : prime_spectrum R) :
+
+/-- The ring homomorphism that takes a section of the structure sheaf of `R` on the open set `U`,
+implemented as a subtype of dependent functions to localizations at prime ideals, and evaluates
+the section on the point corresponding to a given prime ideal. -/
+def open_to_localization (U : opens (Spec.Top R)) (x) (hx : x ∈ U) :
+  (structure_sheaf R).presheaf.obj (op U) ⟶ CommRing.of (localization.at_prime x.as_ideal) :=
+{ to_fun := λ s, (s.1 ⟨x, hx⟩ : _),
+  map_one' := rfl,
+  map_mul' := λ _ _, rfl,
+  map_zero' := rfl,
+  map_add' := λ _ _, rfl }
+
+@[simp] lemma coe_open_to_localization (U : opens (Spec.Top R)) (x) (hx : x ∈ U) :
+  (open_to_localization R U x hx :
+    (structure_sheaf R).presheaf.obj (op U) → localization.at_prime x.as_ideal) =
+  (λ s, (s.1 ⟨x, hx⟩ : _)) :=
+rfl
+
+lemma open_to_localization_apply (U x hx s) :
+  open_to_localization R U x hx s = (s.1 ⟨x, hx⟩ : _) :=
+rfl
+
+/-- The ring homomorphism from the stalk of the structure sheaf of `R` at a point corresponding to
+a prime ideal `p` to the localization of `R` at `p`,
+formed by gluing the `open_to_localization` maps. -/
+def stalk_to_fiber_ring_hom (x : Spec.Top R) :
   (structure_sheaf R).presheaf.stalk x ⟶ CommRing.of (localization.at_prime x.as_ideal) :=
-{ to_fun := (compare_stalks R x).hom ≫ stalk_to_fiber (is_locally_fraction_local R) x,
-  map_zero' := sorry,
-  map_add' := sorry,
-  map_one' := sorry,
-  map_mul' := sorry }
+begin
+  refine limits.colimit.desc _ { X := _, ι := { app := _, naturality' := _, } },
+  { exact λ U, open_to_localization R ((open_nhds.inclusion _).obj (unop U)) x (unop U).2 },
+  { tidy }
+end
 
-/--
-The stalk at `x` is equivalent, as a commutative ring, to the localization at `x`.
--/
-def stalk_iso (x : prime_spectrum R) :
+@[simp] lemma germ_comp_stalk_to_fiber_ring_hom (U x) :
+  (structure_sheaf R).presheaf.germ x ≫ stalk_to_fiber_ring_hom R x =
+  open_to_localization R U x x.2 :=
+limits.colimit.ι_desc _ _
+
+@[simp] lemma stalk_to_fiber_ring_hom_germ' (U : opens (Spec.Top R)) (x) (hx : x ∈ U) (s) :
+  stalk_to_fiber_ring_hom R x ((structure_sheaf R).presheaf.germ ⟨x, hx⟩ s) = (s.1 ⟨x, hx⟩ : _) :=
+ring_hom.ext_iff.1 (germ_comp_stalk_to_fiber_ring_hom R U ⟨x, hx⟩ : _) s
+
+@[simp] lemma stalk_to_fiber_ring_hom_germ (U : opens (Spec.Top R)) (x : U) (s) :
+  stalk_to_fiber_ring_hom R x ((structure_sheaf R).presheaf.germ x s) = s.1 x :=
+by { cases x, exact stalk_to_fiber_ring_hom_germ' R U _ _ _ }
+
+@[simp] lemma to_stalk_comp_stalk_to_fiber_ring_hom (x : Spec.Top R) :
+  to_stalk R x ≫ stalk_to_fiber_ring_hom R x = (localization.of _).to_map :=
+by { erw [to_stalk, category.assoc, germ_comp_stalk_to_fiber_ring_hom], refl }
+
+@[simp] lemma stalk_to_fiber_ring_hom_to_stalk (x : Spec.Top R) (f : R) :
+  stalk_to_fiber_ring_hom R x (to_stalk R x f) = (localization.of _).to_map f :=
+ring_hom.ext_iff.1 (to_stalk_comp_stalk_to_fiber_ring_hom R x) _
+
+-- Switch from widget to plain text before trespassing.
```

Comment:
Does this mean that in widget mode your computer blows up?

Reply:
widget mode takes longer to load

Reply:
I removed, as it seems fine to me. :-)